src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs

```diff
@@ -0,0 +1,98 @@
+            {
+                return null;
+            }
+
+            string baseIntermediatePath = null;
+            var targetFrameworks = new TargetFrameworks();
+            
+            foreach (IProjectVersionedValue<IProjectSubscriptionUpdate> update in updates)
+            {
+                var configurationChanges = update.Value.ProjectChanges[ConfigurationGeneral.SchemaName];
+                baseIntermediatePath = baseIntermediatePath ?? 
+                    configurationChanges.After.Properties[ConfigurationGeneral.BaseIntermediateOutputPathProperty];
+                string targetFrameworkMoniker = 
+                    configurationChanges.After.Properties[ConfigurationGeneral.TargetFrameworkMonikerProperty];
+
+                if (targetFrameworks.Item(targetFrameworkMoniker) == null)
+                {
+                    var projectReferencesChanges = update.Value.ProjectChanges[ProjectReference.SchemaName];
+                    var packageReferencesChanges = update.Value.ProjectChanges[PackageReference.SchemaName];
+
+                    targetFrameworks.Add(new TargetFrameworkInfo
+                    {
+                        TargetFrameworkMoniker = targetFrameworkMoniker,
+                        ProjectReferences = GetProjectReferences(projectReferencesChanges.After.Items),
+                        PackageReferences = GetReferences(packageReferencesChanges.After.Items)
+                    });
+                }
+            }
+
+            return new ProjectRestoreInfo
+            {
+                BaseIntermediatePath = baseIntermediatePath,
+                TargetFrameworks = targetFrameworks
+            };
+        }
+
+        private static IVsReferenceItems GetReferences(IImmutableDictionary<String, IImmutableDictionary<String, String>> items)
+        {
+            return new ReferenceItems(items.Select(p => new ReferenceItem
+            {
+                Name = p.Key,
+                Properties = new ReferenceProperties(p.Value.Select(v => new ReferenceProperty
+                {
+                    Name = v.Key, Value = v.Value
+                })) 
+            }));
+        }
+
+        private static IVsReferenceItems GetProjectReferences(IImmutableDictionary<String, IImmutableDictionary<String, String>> items)
+        {
+            var referenceItems = GetReferences(items);
+            foreach (ReferenceItem item in referenceItems)
+            {
+                var definingProjectDirectory = item.Properties.Item(DefiningProjectDirectoryProperty);
+                string fullPathFromDefining = definingProjectDirectory != null
+                    ? Path.Combine(definingProjectDirectory.Value, item.Name)
+                    : item.Name;
+
+                string projectFileFullPath = Path.GetFullPath(fullPathFromDefining);
+
+                ((ReferenceProperties)item.Properties).Add(new ReferenceProperty
```

Comment:
This does pass all properties through GetReferences above. This methods is just to pass an additional property for Projects

---
In reply to: [83332178](https://github.com/dotnet/roslyn-project-system/pull/524#discussion_r83332178) [](ancestors = 83332178)